corehq/messaging/smsbackends/infobip/models.py

```diff
@@ -103,7 +107,20 @@ def _send_sms(self, config, to, msg, headers):
         }
         url = f'https://{config.personalized_subdomain}.{INFOBIP_DOMAIN}/sms/2/text/advanced'
         response = requests.post(url, json=payload, headers=headers)
-        return response.content
+        self.handle_response(response, msg)
+
+    def handle_response(self, response, msg):
+        if response.status_code != 200:
+            if response.status_code == 500:
```

Comment:
nit: you could have put this as a separate condition outside of the `200` response to be a bit clearer.